go/kbfs/libkbfs/prefetcher.go

```diff
@@ -750,6 +796,14 @@ func (p *blockPrefetcher) run(testSyncCh <-chan struct{}) {
 				continue
 			}
 
+			select {
+			case <-req.obseleted:
+				// This request was cancelled while it was waiting
+				continue
```

Comment:
I think a log message here would be useful in debugging.  Should be pretty rare.